ServantHDBC/src/Api/Photo.hs

```diff
@@ -10,16 +10,24 @@
 module Api.Photo (PhotoAPI, photoHandlers) where
 
 import Control.Monad.Trans.Except
+import Data.Aeson
+import GHC.Generics
 import Servant
 
-type PhotoAPI = "photos" :>
-                (Capture "path_segment_1" String :>
-                 Capture "path_segment_2" String :>
-                 Capture "geometry_or_style" String :>
-                 Capture "original_filename" String :>
-                    Get '[JSON] String)
+type PhotoAPI = "photos" :> PhotoAPI'
 
-photoGet :: String -> String -> String -> String -> ExceptT ServantErr IO String
+type PhotoAPI' =
+                Capture "path_segment_1" String :>
+                Capture "path_segment_2" String :>
+                Capture "geometry_or_style" String :>
+                Capture "original_filename" String :>
```

Comment:
This is probably a misunderstanding on my part. I was trying to follow this part of the spec, 
https://github.com/vacationlabs/haskell-webapps/blob/master/json-api-spec.apib#L393

My belief was, you were interested in seeing how the code would differ when using path vs using query parameters. Path components are unnamed from the API consumer's point of view, but a name is still needed in the handler.

I'm not sure what change you're looking for here.